pieces/intro/rust/src/main.rs

```diff
@@ -139,6 +140,36 @@ fn run7() {
     */
 }
 
+// 8. HashMap
+fn run8() {
+    let mut persons = HashMap::new();
+    persons.insert("youngest", Person::new("Samantha", 19));
+    persons.insert("middle", Person::new("Wade", 20));
+    persons.insert("oldest", Person::new("James", 65));
+
+    println!("All people: {:?}", persons);
+    println!("Youngest person: {:?}", persons.get("youngest"));
+    /*
+    All people: {"youngest": Person: Samantha, 19, "middle": Person: Wade, 20, "oldest": Person: James, 65}
+    Youngest person: Some(Person: Samantha, 19)
```

Comment:
The reason the Rust output is different than the Python out here is because the result of the `get` is an option, `Some<Person>`. I considered making them the same by using `get` in python also, but that would require using an `if let` here. Something like

```rs
if let Some(youngest) = persons.get("youngest") {
    println!("Youngest person: {youngest}");
else {
    println!("No youngest person found");
}
```

This seemed too complicated for an introduction. Do you agree?

Reply:
Hmmm. I was wondering if we could avoid instantiating a `Person` in either case and instead demonstrate the ability of a `dict` and a `HashMap` as a lookup containers? 

Something like this:

- A product ID to product name mapping
- Lookup by product ID (key) and if possible, look up by product name (value)
- Similar example for sets, but with just product names showing there are no duplicates
- Look up by product name in set

Reply:
It does deviate from the general theme of "persons" in the intro, but I couldn't think of a good way to demonstrate lookups using person objects 😅

Reply:
Got it, that makes sense and I should be able to get them to look mostly the same that way. I'll work on updating it.

Reply:
See if you think these new examples will work better. I changed from people to processors.